server/session.go

```diff
@@ -255,12 +257,24 @@ func (s *Session) isCluster() bool {
 	return s.isProxy() || s.isMultiplex()
 }
 
+func (s *Session) unlockTerminating() {
+	if s == nil {
```

Comment:
I don't see why this is needed. The `s == nil` is always checked before locking, and `nil` is never assigned to `s` while it's locked.

Reply:
Actually, you are right. Thanks.
I didn't cleared up how `defer` works for myself. Code changed.